librz/util/thread_iterators.c

```diff
@@ -52,8 +52,8 @@ static void *thread_iterate_list_cb(th_list_ctx_t *context) {
 		context->head = rz_list_iter_get_next(context->head);
 		rz_th_lock_leave(lock);
 
-		if (element) {
-			iterator(element, user);
+		if (element && !iterator(element, user)) {
+			break;
```

Comment:
you cannot do this. it will kill only one thread, not all of them.

Reply:
This also breaks compatibility between rz_th_iterate_list and rz_th_iterate_pvector

Reply:
Mh, this was a change copied from your search branch. What was the intention back then?

Reply:
If i wrote that, i must have been under heavy drugs. We can close this.

Reply:
![image](https://github.com/user-attachments/assets/6fb63a45-4ef5-4eee-865b-237abeb9eba6)
Just checked. this is the correct patch. i do not know where that comes from

Reply:
Not this specific line. Just the return value. Just added it here because otherwise the change of the return value would have been ignored? Or was it meant as a general way to signal if the iterator finished properly?

Reply:
When you write thread stuff you need to consider that the stack duplicated (thus not shared) for each thread context.
Threads needs syncronization mechanisms when you need to stop all of them, and this is usually done via notification system for this or a shared value.

rz_atomic_XXX does this, via shared value. we have a thread safe shared value (a boolean) which we use to ensure a thread can be stopped by any thread that fails. why we need a thread safe value? because depending on the CPU cache, a thread might see a non updated value.

With the original patch, if you had 10 threads, and you fail in one, the other 9 will keep running and the failed thread will just halt.